kie-remote/src/main/java/org/kie/remote/impl/RemoteKieSessionImpl.java

```diff
@@ -40,6 +46,21 @@ public void close() {
 
     @Override
     public RemoteEntryPoint getEntryPoint( String name ) {
-        return getEntryPoint(name);
+        return entryPoints.computeIfAbsent( name, k -> new RemoteEntryPointImpl(sender, k, topicsConfig, requestsStore, delegate) );
```

Comment:
If this could be a concurrent hash map in the future, please note that Java 8 had two concurrency bugs in ConcurrentHashMap's computeIfAbsent().